src/app/CommandHandler.h

```diff
@@ -332,6 +342,16 @@ class CommandHandler
      * When refcount reached 0, CommandHandler will send the response to the peer and shutdown.
      */
     virtual void DecrementHoldOff(Handle * apHandle) {}
+
+    /**
+     * Returns the ExchangeContext, if one is still available, for use during asynchronous
+     * command processing. This is a best-effort accessor with no guarantees that
+     * an ExchangeContext is present once a command has gone async.
+     *
+     * This method exists to prevent use of GetExchangeContext() in async code paths and
+     * must NOT be used by cluster implementations.
+     */
+    virtual Messaging::ExchangeContext * TryGetExchangeContextWhenAsync() const { return nullptr; };
```

Comment:
Can you just make this private?

Reply:
If private we can clean comments a bit (and it is obvious then that cluster implementations will not use it).

I would still welcome an explanation of what it needs to be used safely.

Reply:
OK, I guess the key part is "not public".